src/VecSim/algorithms/hnsw/visited_list_pool.h

```diff
@@ -0,0 +1,77 @@
+#pragma once
+
+#include <mutex>
+#include <string.h>
+
+namespace hnswlib {
+typedef unsigned short int vl_type;
+
+class VisitedList {
+  public:
+    vl_type curV;
+    vl_type *mass;
+    unsigned int numelements;
+
+    VisitedList(int numelements1) {
+        curV = -1;
+        numelements = numelements1;
+        mass = new vl_type[numelements];
+    }
+
+    void reset() {
+        curV++;
+        if (curV == 0) {
+            memset(mass, 0, sizeof(vl_type) * numelements);
+            curV++;
+        }
+    };
+
+    ~VisitedList() { delete[] mass; }
+};
+///////////////////////////////////////////////////////////
+//
+// Class for multi-threaded pool-management of VisitedLists
+//
+/////////////////////////////////////////////////////////
+
+class VisitedListPool {
```

Comment:
This is hnsw implementation for multi-threaded insert/search. So actually we don't need the pool logic in single-threaded mode, I'll change the code so that it will be under ENABLE_PARALLELISATION build.